### PR TITLE
Remove duplicate Publications in response

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/DynamoDBPublicationService.java [hunk 1]
```diff
@@ -227,7 +227,8 @@ private void validateIdentifier(UUID identifier, Publication publication) throws
             throw new DynamoDBException(ERROR_READING_FROM_TABLE, e);
         }
 
-        return parseJsonToPublicationSummaries(items);
+        List<PublicationSummary> publications = parseJsonToPublicationSummaries(items);
+        return filterOutOlderVersionsOfPublications(publications);
```
Comment (axthosarouris) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/DynamoDBPublicationService.java:231]: Could we have a test for that?

**Review by axthosarouris — COMMENTED** [review]
test missing?

**Review by brinxmat — COMMENTED** [review]
+1 on axthosarouris' comment

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/DynamoDBPublicationService.java (hunk 1 above)
Comment (janbj) [reply publication-commons/src/main/java/no/unit/nva/publication/service/impl/DynamoDBPublicationService.java:231]: The method already has tests.

Comment (axthosarouris) [reply publication-commons/src/main/java/no/unit/nva/publication/service/impl/DynamoDBPublicationService.java:231]: I actually this is the problem which explains why `filterOutOlderVersionsOfPublications` is protected and not private. The module is the DynamoDBPublicationService and it is through the public methods of this module that we need to test the module functionality. 
The fact that we changed the functionality of a public method and no tests failed and no tests were changed is problematic.
`filterOutOlderVersionsOfPublications` is an implementation detail. This method should be private and there should be a test with a name somethiing like `listPublishedPublicationsByDateReturnsTheMostRecentVersionOfEachPublication`

**Review by brinxmat — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**